### PR TITLE
Set up infrastructure to trigger help center tips

### LINKED ISSUE #6633: Set up infrastructure to interact with Help Center
## Feature Description

Ahead of automagically opening FTUE when elements go outside the safezone we need the infrastructure to have this event (and others to follow) occur in a way that is uniform and testable. 
This ticket is for that foundation. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by embarks**
## Context
We have a help center which shows users tips on how to use the editor. The help center should open and show tips relevant to the user's context.
The in-context experience is described in [the figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=6515%3A522898).

## Summary
Create a new module `useUserOnboarding` to handle the following:
1. only show the relevant tip if it is the user's first time in the context
2. Create a map of triggers to help center actions and perform the help center actions

## Relevant Technical Choices
- Moved some help center logic and use providers to share the actions which can open and control the help center from another context
- Created a new module `useUserOnboarding` to contain common onboarding events that components and hooks can easily access with a key/selector
- ~[todo] Persist new keys to the `web_stories_onboarding.meta` property for the `currentUser` to determine if they have been onboarded or not (similar to persisting which help center tips a user has read)~
- Since we only show the tip if it's not read, we can use that persistence for this first FTUE feature.

## To-do
- [x] polish and publish this pull request
- [x] when the safe zone/design guideline is visible to the user for the first time, open the help center and show the relevant quick tip

## User-facing changes
- Now when a user moves an element outside the safe zone for the first time and the relevant quick tip is not read, open the help center menu and show the user that tip

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Clear your read tips in your console by typing `await wp.apiFetch({ path: '/web-stories/v1/users/me', method: 'POST', data: { meta: { web_stories_onboarding: {} }}})` when in the editor
2. Create a new story and add an element or template
3. Move an element so that the safe zone appears
4. The help center menu should open to the Safe Zone tip
5. Close the menu
6. Move another element so the safe zone appears
7. The menu should remain closed

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6633 and #6368

### CODE REVIEW

File: assets/src/edit-story/editorApp.js [hunk 1]
```diff
@@ -70,14 +71,16 @@ function App({ config }) {
...
                                   <DropTargetsProvider>
-                                    <GlobalStyle />
-                                    <DevTools />
-                                    <DefaultMoveableGlobalStyle />
-                                    <CropMoveableGlobalStyle />
-                                    <ModalGlobalStyle />
-                                    <CalendarStyle />
-                                    <KeyboardOnlyOutlines />
-                                    <Layout />
+                                    <HelpCenterProvider>
```
Comment (maxyinger) [inline assets/src/edit-story/editorApp.js:76]: We should be good here with updates since this is using useContextSelector under the hood, but something to keep an eye on cus any app wide rerenders could cause some jank in the transitions

File: assets/src/edit-story/app/userOnboarding/useUserOnboarding.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  // const { updateCurrentUser, currentUser } = useCurrentUser(
+  //   ({ state, actions }) => ({
+  //     currentUser: state.currentUser,
+  //     updateCurrentUser: actions.updateCurrentUser,
+  //   })
+  // );
+
+  // const { goToTip } = useHelpCenter(({ actions }) => ({
+  //   goToTip: actions.goToTip,
+  // }));
```
Comment (maxyinger) [inline assets/src/edit-story/app/userOnboarding/useUserOnboarding.js]: I think we'll need to create a dedicated action in the help center that's like `openUpToTipIfUnread(TIP_KEY)` (pardon my inability to come up with a good name lol 😅).

Only reason I say that is I think the more apparent things it will do are opening up the modal and setting the current tip index on the same render as the modal opening (so we don't see it transition to the tip as it's opening), but also to set some sort of flag so that a lot of the inference and reseting effects are opted out of on that specific action.

basically have it return some state partial like this
```js
(tipKey) => isTipKeyUnread(tipKey) && ({
  isOpen: true,
  navigationIndex: getIndexFromKey(tipKey),
  // some flag like this to opt out of any resetting or smart behavior we have in the effects
  isOpeningToIndex: true
})
```

File: assets/src/edit-story/app/userOnboarding/useUserOnboarding.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Internal dependencies
+ */
+// import { useCurrentUser } from '../currentUser';
+// import { useHelpCenter } from '../helpCenter';
+
+const TRIGGER_CONTEXTS = {
+  SAFE_ZONE: 'safeZone',
+  // todo [@embarks]: other onboarding triggers
+};
```
Comment (maxyinger) [inline assets/src/edit-story/app/userOnboarding/useUserOnboarding.js]: I'm wondering if these triggers are different than just using the tips data constant from the help center?

Will these triggers always map 1:1 to opening up a single tip in the help center? or are there other scenarios that wouldn't be so 1:1 and necessitate extra data that maps to the tip data?

**Review by maxyinger — APPROVED** [review]
Looks great so far! Adding context to the `useHelpCenter` hook like you did was exactly what I was thinking.

I guess the only part I'm not following on is are the `useUserOnboarding` triggers different from a tip being read?

My initial assumption was just that all the data we'd need persisted was just the read and unread tips. If that's the case I think we can just create a specific action in the help center that opens the modal to a tip if the user hasn't already read it and just expose that.

However if the feature necessitates extra data bookkeeping on top of the tips, totally agree that the `useUserOnboarding` is a nice layer of abstraction there. I guess also if you were thinking of baking tips into a `useHelpCenter` action and just returning a no argument action from `useUserOnboarding`, that makes sense too (from the looks of it, that's where this is heading and that totally works too)

File: assets/src/edit-story/app/userOnboarding/useUserOnboarding.js (hunk 3 above)
Comment (embarks) [reply assets/src/edit-story/app/userOnboarding/useUserOnboarding.js]: This is a great question. What I was thinking was we might have onboarding steps that are not related to the help center. We might want to trigger a highlight, or some other step-based onboarding, or track the items separately. But I might be pre-optimizing and I think if all they do is open a tip we should just use the same constants.

But I think I need to open up a discussion about what the UX should be around these triggers. Right now it's variable what in-context triggers open which tip. For masking, for example, the ratio is 4:1 for in-context-triggers to tips. So I need to know if we need to track an in-context trigger event independently from whether the user has read the tip.

File: assets/src/edit-story/editorApp.js (hunk 1 above)
Comment (embarks) [reply assets/src/edit-story/editorApp.js:76]: I was thinking about your [post about stores](https://formidable.com/blog/2021/stores-no-context-api/) while writing this lol

Comment (maxyinger) [reply assets/src/edit-story/editorApp.js:76]: haah yea would have been real convenient to expose this to the rest of the app with one

File: assets/src/edit-story/app/userOnboarding/useUserOnboarding.js (hunk 3 above)
Comment (maxyinger) [reply assets/src/edit-story/app/userOnboarding/useUserOnboarding.js]: Oh yea, didn't think about having other actions than opening the help center in here. good call!

**Review by BrittanyIRL — APPROVED** [review]
Awesome! Thank you for the details on how to test. Like the updates you made a lot and works how it should! 🚀

**Review by maxyinger — APPROVED** [review]
looks great! very clean, and I like how you set up `useOnboarding` to give us a nice foundation to handle actions beyond just triggering FTUE
